packages/babel-core/src/util.js

```diff
@@ -111,20 +126,22 @@ export function booleanify(val: any): boolean | any {
 
 export function shouldIgnore(
```

Comment:
The changes to this function make it required that you both match `only` and don't match `ignore`. With the old logic of only using `only` when present meant that we needed to aggregate the whole config tree before processing `only` and `ignore`. Like say you had

    babel.transform(..., {ignore: "**/*.jsx"});

but your `.babelrc` was

    { 
      only: ['**/*.jsx'],
    }

in v6 the logic would compile all `.jsx` files because it would pass `only`, and we're required to read the `.babelrc` of _every_ file even if they won't match the `ignore` pattern passed to `.transform`.